### PR TITLE
Improve library destination detection and fix version.txt issues

### LINKED ISSUE #9: Application Directory Detection is broken under linux
Tested on 0.0.1.
image

### PR DESCRIPTION
**Pull request by NyCodeGHG**
This PR fixes #9﻿

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -70,13 +70,15 @@ tasks {
             Files.createDirectories(parent)
         }
 
-        val version = project.version
-        val buildId = System.getenv("GITHUB_RUN_ID") ?: "<local build>"
-        val commitHash = System.getenv("GITHUB_SHA") ?: gitCommitHash()
+        doLast {
```
Comment (DRSchlaubi) [inline build.gradle.kts:73]: What does the doLast do here?

File: lib/utils/platform_utils/chmod_lib.dart [hunk 2]
```diff
@@ -50,16 +50,16 @@ DynamicLibrary? _load() {
 
   String programDirectory;
   // Check if we are in a development environment
-  if (!Platform.resolvedExecutable.endsWith('dart') &&
-      !Platform.resolvedExecutable.endsWith('dart.exe')) {
-    programDirectory = path.dirname(Platform.resolvedExecutable);
+  if (!Platform.resolvedExecutable.endsWith('dart')) {
+    final scriptLocation = Platform.resolvedExecutable.replaceAll('\\', '/');
```
Comment (DRSchlaubi) [inline lib/utils/platform_utils/chmod_lib.dart:54]: Put this into a util so we don't have the same logic multiple times

**Review by DRSchlaubi — CHANGES_REQUESTED** [review]
Did you test this on windows?

File: build.gradle.kts (hunk 1 above)
Comment (NyCodeGHG) [reply build.gradle.kts:73]: doLast ensures that the code is executed only when the task is executed and not when the buildscript is evaluated.

File: bin/mcserv.dart [hunk 3]
```diff
@@ -100,8 +100,8 @@ void main(List<String> arguments) async {
 Future<void> _version() async {
   final scriptLocation = Platform.resolvedExecutable.replaceAll('\\', '/');
   final mcServInstall =
-      scriptLocation.substring(0, scriptLocation.lastIndexOf('/')).split('/');
-  final versionFile = fs.file(path.joinAll([...mcServInstall, 'version.txt']));
+      scriptLocation.substring(0, scriptLocation.lastIndexOf('/'));
+  final versionFile = fs.file(path.join(mcServInstall, 'version.txt'));
```
Comment (DRSchlaubi) [inline bin/mcserv.dart:103]: Splitting by `/(?<!^)\/` might be better as it does more what joinAll expects

**Comment by NyCodeGHG** [comment]
It's working on windows
image

**Review by DRSchlaubi — APPROVED** [review]

**Pull request merged**